applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp

```diff
@@ -0,0 +1,212 @@
+            {
+                    if (r_N(point_number, i) > shape_function_tolerance) {
+                        for (IndexType j = 0; j < number_of_nodes; j++) // loop over shape functions of displacements
+                        {
+                                if (r_N(point_number, j) > shape_function_tolerance) {
+                                    const double NN = r_N(point_number, j) * r_N(point_number, i);
+
+                                    const IndexType ibase = counter_n * 3 + 3 * (number_of_non_zero_nodes);
+                                    const IndexType jbase = counter_m * 3;
+
+                                    // Matrix in following shape:
+                                    // |0 H^T|
+                                    // |H 0  |
+
+                                    LHS(ibase, jbase) = NN;
+                                    LHS(ibase + 1, jbase + 1) = NN;
+                                    LHS(ibase + 2, jbase + 2) = NN;
+
+                                    LHS(jbase, ibase) = NN;
+                                    LHS(jbase + 1, ibase + 1) = NN;
+                                    LHS(jbase + 2, ibase + 2) = NN;
+
+                                    counter_m++;
+                                }
+                        }
+                        counter_n++;
+                }
+                if (CalculateStiffnessMatrixFlag) {
+                    noalias(rLeftHandSideMatrix) += LHS;
+                }
+
+                if (CalculateResidualVectorFlag) {
+                    const array_1d<double, 3>& displacement = (Has(DISPLACEMENT))
+                        ? this->GetValue(DISPLACEMENT)
+                        : ZeroVector(3);
+
+                    IndexType counter = 0;
+                    for (IndexType i = 0; i < number_of_nodes; i++)
+                    {
+                        for (IndexType n = 0; n < r_N.size1(); ++n) {
+                            if (r_N(n, i) > shape_function_tolerance) {
+                                const array_1d<double, 3>& disp = r_geometry[i].FastGetSolutionStepValue(DISPLACEMENT);
+
+                                IndexType index = 3 * counter;
+                                u[index]     = (disp[0] - displacement[0]);
+                                u[index + 1] = (disp[1] - displacement[1]);
+                                u[index + 2] = (disp[2] - displacement[2]);
+                                counter++;
+                            }
+                        }
+                    }
+                    for (IndexType i = 0; i < number_of_nodes; i++)
+                    {
+                        for (IndexType n = 0; n < r_N.size1(); ++n) {
+                            if (r_N(n, i) > shape_function_tolerance) {
+                                const array_1d<double, 3>& l_m = r_geometry[i].FastGetSolutionStepValue(VECTOR_LAGRANGE_MULTIPLIER_REACTION);

```

Comment:
@rickyaristio the variable is introduced only to enable the computation with the reactions. At this position the core variable VECTOR_LAGRANGE_MULTIPLIER makes more sense.

Reply:
Okay, that was a misunderstanding.